lib/ansible/modules/windows/setup.ps1

```diff
@@ -307,7 +307,12 @@ if($gather_subset.Contains('memory')) {
 if($gather_subset.Contains('platform')) {
     $win32_cs = Get-LazyCimInstance Win32_ComputerSystem
     $win32_os = Get-LazyCimInstance Win32_OperatingSystem
-    $ip_props = [System.Net.NetworkInformation.IPGlobalProperties]::GetIPGlobalProperties()
+    $domain_suffix = $win32_cs.Domain.Substring($win32_cs.Workgroup.length)
```

Comment:
Now `ansible_domain` will fail in a workgroup because `$domain_suffix` is an empty string and `.Substring(1)` will fail. What I suggest you do is;

```suggestion
    $domain = $win32_cs.Domain.Substring($win32_cs.Workgroup.length)
    $fqdn = $win32_cs.DNSHostname
    if ($domain -ne "") {
        $fqdn += ".$domain"
    }
    
    try {
        $ansible_reboot_pending = Get-PendingRebootStatus
    } catch {
        # fails for non-admin users, set to null in this case
        $ansible_reboot_pending = $null
    }

    $ansible_facts += @{
        ansible_architecture = $win32_os.OSArchitecture
        ansible_domain = $domain
        ansible_fqdn = $fqdn
        ansible_hostname = $win32_cs.DNSHostname
        ansible_netbios_name = $win32_cs.Name
        ansible_kernel = $osversion.Version.ToString()
        ansible_nodename = $fqdn
```

What this means is;

* `ansible_domain`: Is the value returned from `$win32_cs.Domain.Substring($win32_cs.Workgroup.length)` which in a domain environment is the domain part and a workgroup is an empty string
* `ansible_fqdn`: Is just the DNS hostname in a workgroup, if a domain is present it is the DNS hostname with `.$domain` appended
* `ansible_nodename`: Is the same as fqdn

This solves the issue where `ansible_domain` is an empty string for a workgroup and contains the domain part without the leading `.` for an actual domain environment. Also the fqdn will not contain the `.` suffix in a workgroup as we only add that if the domain value is not an empty string.

_Note; I'm not sure how well GitHub suggestions work for multiple lines, you may need to manually apply this, or modify it how you like._